python/hail/dataset.py

```diff
@@ -1019,181 +886,162 @@ def annotate_variants_expr(self, expr):
-        annotation ``va.gene`` is set to the gene name occurring in the fifth
-        column and ``NA`` otherwise.
-
-        >>> vds_result = vds.annotate_variants_intervals('data/exons2.interval_list', 'va.gene')
+        ``expr`` has the following symbols in scope:
 
-        **Notes**
+          - ``va``: variant annotations
+          - ``table``: See note.
 
-        There are two formats for interval list files.  The first appears as
-        ``chromosome:start-end`` as in the first example.  This format will
-        annotate variants with a *Boolean*, which is ``true`` if that variant is
-        found in any interval specified in the file and `false` otherwise.
+        .. note:: 
+        
+            The value of ``table`` inside root/expr depends on the number of values in the key table, 
+            as well as the ``product`` argument. There are three behaviors based on the number of values
+            and one branch for ``product`` being true and false, for a total of six modes:
+            
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | Number of value columns | ``product`` | Type of  ``table`` | Value of  ``table``                           |
+            +=========================+=============+====================+===============================================+
+            | More than 2             | False       | ``Struct``         | Struct with an element for each column.       |
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | 1                       | False       | ``T``              | The value column.                             |
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | 0                       | False       | ``Boolean``        | Existence of any matching key.                |
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | More than 2             | True        | ``Array[Struct]``  | An array with a struct for each matching key. |
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | 1                       | True        | ``Array[T]``       | An array with a value for each matching key.  |
+            +-------------------------+-------------+--------------------+-----------------------------------------------+
+            | 0                       | True        | ``Int``            | The number of matching keys.                  |
+            +-------------------------+-------------+--------------------+-----------------------------------------------+  
+                      
+        **Common uses for the** ``expr`` **argument**
```

Comment:
Can you double-check this bit is formatting correctly?  It looks weird to me (some code blocks missing?)